src/frontend/src/common/components/library/data_subview/components/DownloadModal/index.tsx

```diff
@@ -0,0 +1,304 @@
+import { Dialog } from "@material-ui/core";
+import CloseIcon from "@material-ui/icons/Close";
+import { Alert, Tooltip } from "czifui";
+import React, { useEffect, useState } from "react";
+import { CSVLink } from "react-csv";
+import { useMutation } from "react-query";
+import DialogActions from "src/common/components/library/Dialog/components/DialogActions";
+import DialogContent from "src/common/components/library/Dialog/components/DialogContent";
+import DialogTitle from "src/common/components/library/Dialog/components/DialogTitle";
+import { useUserInfo } from "src/common/queries/auth";
+import { downloadSamplesFasta } from "src/common/queries/samples";
+import { TooltipDescriptionText, TooltipHeaderText } from "../../style";
+import {
+  CheckBoxInfo,
+  CheckBoxWrapper,
+  Container,
+  Content,
+  DownloadType,
+  DownloadTypeInfo,
+  Header,
+  StyledButton,
+  StyledCheckbox,
+  StyledIconButton,
+  Title,
+} from "./style";
+
+interface Props {
+  sampleIds: string[];
+  failedSamples: any[];
+  isMetadataSelected: boolean;
+  setMetadataSelected: (prevState: boolean) => void;
+  isFastaSelected: boolean;
+  setFastaSelected: (prevState: boolean) => void;
+  isFastaDisabled: boolean;
+  setFastaDisabled: (prevState: boolean) => void;
+  tsvData: [string[], string[][]] | undefined;
+  open: boolean;
+  onClose: () => void;
+  setDownloadFailed: (hasFailed: boolean) => void;
+}
+
+const DownloadModal = ({
+  sampleIds,
+  failedSamples,
+  isMetadataSelected,
+  setMetadataSelected,
+  isFastaSelected,
+  setFastaSelected,
+  isFastaDisabled,
+  setFastaDisabled,
+  tsvData,
+  open,
+  onClose,
+  setDownloadFailed,
+}: Props): JSX.Element => {
+  const { data } = useUserInfo();
+  const groupName = data?.group?.name.toLowerCase().replace(/ /g, "_"); // format group name for sequences download file
+  const downloadDate = new Date();
+  const separator = "\t";
+  const [tsvRows, setTsvRows] = useState<string[][]>([]);
+  const [tsvHeaders, setTsvHeaders] = useState<string[]>([]);
+
+  useEffect(() => {
+    if (tsvData) {
+      const [Headers, Rows] = tsvData;
+      setTsvHeaders(Headers);
+      setTsvRows(Rows);
+    }
+  }, [tsvData]);
+
+  useEffect(() => {
+    if (JSON.stringify(sampleIds) === JSON.stringify(failedSamples)) {
+      setFastaDisabled(true);
+    } else {
+      setFastaDisabled(false);
+    }
+  }, [sampleIds, failedSamples]);
+
+  const handleMetadataClick = function () {
+    setMetadataSelected(!isMetadataSelected);
+  };
+
+  const handleFastaClick = function () {
+    setFastaSelected(!isFastaSelected);
+  };
+
+  const mutation = useMutation(downloadSamplesFasta, {
+    onError: () => {
+      setDownloadFailed(true);
+    },
+    onSuccess: (data: any) => {
+      const link = document.createElement("a");
+      link.href = window.URL.createObjectURL(data);
+      link.download = `${groupName}_sample_sequences_${downloadDate
+        .toISOString()
+        .slice(0, 10)}.fasta`;
+      link.click();
```

Comment:
maybe remove the link element after link to prevent memory bloat and DOM pollution 😄 ?

```ts
...
link.click()
link.remove()
```

Reply:
oh yes! Did not realize that would pollute the DOM!!